v2/tools/generator/internal/config/multi_matcher.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// MultiMatcher is a matcher that contains multiple other matchers
+type MultiMatcher struct {
+	matchers []StringMatcher
+}
+
+var _ StringMatcher = &MultiMatcher{}
+
+// newMultiMatcher returns a new matcher for multiple strings
+func newMultiMatcher(matcher string) *MultiMatcher {
+	var matchers []StringMatcher
+	for _, str := range strings.Split(matcher, ";") {
+		m := NewStringMatcher(str)
+		matchers = append(matchers, m)
+	}
+
+	return &MultiMatcher{
+		matchers: matchers,
+	}
+}
+
+// Matches returns true if the provided value is matched by any
+func (mm *MultiMatcher) Matches(value string) bool {
+	result := false
+	// MUST not short circuit - for the suggestions generated by any nested matchers to be useful,
+	// each matcher must see all potential match candidates.
+	for _, m := range mm.matchers {
+		if m.Matches(value) {
+			result = true
+		}
+	}
+
+	return result
+}
+
+// WasMatched returns nil if every nested matcher had a match, otherwise returns a diagnostic error
+func (mm *MultiMatcher) WasMatched() error {
+	var errs []error
+	count := 0
+	for _, m := range mm.matchers {
+		if err := m.WasMatched(); err != nil {
+			errs = append(errs, err)
+			count++
+		}
+	}
+
+	err := kerrors.NewAggregate(errs)
+	if err != nil {
+		return errors.Wrapf(err, "%d of %d did not match", count, len(mm.matchers))
+	}
+
+	return nil
```

Comment:
`kerrors.NewAggregate` and `errors.Wrapf` both return nil if there are no errors (and `Wrapf` avoids formatting in that case as well), so you _could_ do:

```suggestion
	return errors.Wrapf(kerrors.NewAggregate(errs), "%d of %d did not match", count, len(mm.matchers))
```

Reply:
Good suggestion; adopted with formatting changes.